NuGet.Services.Dashboard.Operations/Infrastructure/CommandLineParser.cs

```diff
@@ -147,7 +154,10 @@ public ICommand ParseCommandLine(IEnumerable<string> commandLineArgs)
                 throw new CommandLineException(TaskResources.UnknownCommandError, cmdName);
             }
 
-            ExtractOptions(cmd, argsEnumerator);
+            var secretReader = _secretReaderFactory.CreateSecretReader();
+            var secretInjector = _secretReaderFactory.CreateSecretInjector(secretReader);
```

Comment:
no need to create a secretReader every time the method is called (it's expensive). Add a Lazy<ISecretInjector>, and have it initialized in the ctor.